### PR TITLE
Add Datadog Agent

### PR DESCRIPTION
**Pull request by g-linville**
This adds the Datadog Agent to the Hub System Charts.

### CODE REVIEW

File: templates/datadog-agent.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+kind: HelmChart
+metadata:
+  name: datadog-agent
+  namespace: kube-system
+spec:
+  bootstrap: true
+  chart: https://github.com/DataDog/helm-charts/releases/download/datadog-3.25.0/datadog-3.25.0.tgz
+  targetNamespace: datadog-agent
+  valuesContent: |
+    datadog:
```
Comment (StrongMonkey) [inline templates/datadog-agent.yaml:28]: Do you only prefer to run this in user nodes? If so, you might need to toleration this taint:

```
https://github.com/acorn-io/hub-system-charts/blob/main/templates/traefik.yaml#L75
```

File: templates/datadog-agent.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  labels:
+    release: datadog-agent
+    istio-injection: disabled
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: datadog-api-key
+  namespace: datadog-agent
+stringData:
```
Comment (StrongMonkey) [inline templates/datadog-agent.yaml:15]: Interesting, I am not sure if stringData works well in helm chart. You can also use base64 enc function in helm.

File: templates/datadog-agent.yaml (hunk 1 above)
Comment (g-linville) [reply templates/datadog-agent.yaml:28]: Good call

**Comment by g-linville** [comment]
Fixed both @StrongMonkey

**Review by StrongMonkey — APPROVED** [review]

**Pull request merged**